WorkloadManagementSystem/Agent/SiteDirector.py

```diff
@@ -1341,6 +1305,70 @@ def updatePilotStatus(self):
 
     return S_OK()
 
+  def _updatePilotStatusPerQueue(self, queue):
+    """ Update status of pilots in transient state for a given queue
+
+    :param str queue: queue name
+    """
+    ce = self.queueDict[queue]['CE']
+    ceName = self.queueDict[queue]['CEName']
+    queueName = self.queueDict[queue]['QueueName']
+    ceType = self.queueDict[queue]['CEType']
+    siteName = self.queueDict[queue]['Site']
+
+    result = pilotAgentsDB.selectPilots({'DestinationSite': ceName,
+                                         'Queue': queueName,
+                                         'GridType': ceType,
+                                         'GridSite': siteName,
+                                         'Status': TRANSIENT_PILOT_STATUS,
+                                         'OwnerDN': self.pilotDN,
+                                         'OwnerGroup': self.pilotGroup})
+    if not result['OK']:
+      self.log.error('Failed to select pilots", ": %s' % result['Message'])
+      return
+    pilotRefs = result['Value']
+    if not pilotRefs:
+      return
+
+    result = pilotAgentsDB.getPilotInfo(pilotRefs)
+    if not result['OK']:
+      self.log.error('Failed to get pilots info from DB', result['Message'])
+      return
+    pilotDict = result['Value']
+
+    stampedPilotRefs = []
+    for pRef in pilotDict:
+      if pilotDict[pRef]['PilotStamp']:
+        stampedPilotRefs.append(pRef + ":::" + pilotDict[pRef]['PilotStamp'])
+      else:
+        stampedPilotRefs = list(pilotRefs)
+        break
+
+    # This proxy is used for checking the pilot status and renewals
+    # We really need at least a few hours otherwise the renewed
+    # proxy may expire before we check again...
+    result = ce.isProxyValid(3 * 3600)
+    if not result['OK']:
+      result = gProxyManager.getPilotProxyFromDIRACGroup(self.pilotDN, self.pilotGroup, 23400)
```

Comment:
The SiteDirector is implemented such that it works for one VO only. Therefore, the pilotDN/group is the same for all the pilots. This can be set up once for all the queues. So, it can be done just before feeding the ThreadPool

Reply:
@aldbr can you react to this?

Reply:
@atsareg is there any hidden association between the `requiredTimeLeft` value (23400) and the `valid` value (23300) in the `setProxy` method on the following lines?
https://github.com/DIRACGrid/DIRAC/pull/4338/files/2b338f765f0aa3671b92b419a0c6478bd49f7942#diff-ed886d0dfc6dcbc6b62699e32f11b07bR1352

https://github.com/DIRACGrid/DIRAC/pull/4338/files/2b338f765f0aa3671b92b419a0c6478bd49f7942#diff-ed886d0dfc6dcbc6b62699e32f11b07bR1356

Can I safely move the `getPilotProxyFromDIRACGroup` method without impacting the `setProxy` method?

Reply:
@aldbr was `getPilotProxyFromDIRACGroup` moved?